pyquil/quil.py

```diff
@@ -138,7 +161,7 @@ def instructions(self) -> List[AbstractInstruction]:
 
         return self._synthesized_instructions
```

Comment:
adding an `assert self._synthesized_instructions is not None` on the previous line resolves the following mypy error:

```
pyquil/quil.py:162: error: Incompatible return value type (got "Optional[List[AbstractInstruction]]", expected "List[AbstractInstruction]")
```

Reply:
Will appear in the 8th commit.